### PR TITLE
OCPBUGS-8215: bugfix in Node Exporter argument setting

### PR DESCRIPTION
**Pull request by raptorsun**
* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

This is the fix to the bug found by @juzhao  in https://github.com/openshift/cluster-monitoring-operator/pull/1905#issuecomment-1451186414

### CODE REVIEW

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Review by simonpasquier — COMMENTED** [review]
This breaks the unit tests.

**Comment by raptorsun** [comment]
> This breaks the unit tests.

Unit test are fixed now :)

**Comment by juzhao** [comment]
issue is fixed, details see from bug comment
/label qe-approved

**Comment by raptorsun** [comment]
> /retest-required

I find out why the test `e2e-agnostic-operator` fails, it misses the metric `node_network_speed_bytes` because the test node does not have the `ethtool` netlink interface. I have created a new PR to fix this by disabling netlink mode of netclass collector by default. I suggest we use the new PR to fix this bug.
https://github.com/openshift/cluster-monitoring-operator/pull/1911

File: pkg/manifests/manifests.go [hunk 1]
```diff
@@ -2140,7 +2140,8 @@ func (f *Factory) setTLSSecurityConfiguration(args []string, tlsCipherSuitesArg
 func setArg(args []string, argName string, argValue string) []string {
 	found := false
 	for i, arg := range args {
-		if strings.HasPrefix(arg, argName) {
+		if arg == argName ||
+			(len(argName) > 0 && argName[len(argName)-1] == '=' && strings.HasPrefix(arg, argName)) {
```
Comment (simonpasquier) [inline pkg/manifests/manifests.go:2144]: Is there a case where `argName` can be empty or is it just by precaution?

Comment (raptorsun) [reply pkg/manifests/manifests.go:2144]: this is by precaution, preventing a potential out of index error :)

**Comment by raptorsun** [comment]
The PR to fix the failling test `e2e-agnostic-operator` is https://github.com/openshift/cluster-monitoring-operator/pull/1912 . 
That PR#1912 should merge  before this one.

**Comment by raptorsun** [comment]
Node exporter cannot collect the metric `node_network_speed_bytes` in the CI cluster, but in a cluster created by cluster bot the metric `node_network_speed_bytes` can be scraped. Still checking where comes such discrepancy  🤔

**Comment by raptorsun** [comment]
The cluster created by cluster bot on GCP is able to emit `speed` metric in `/host/root/sys/class/net/*/`, 
while the `speed` file of a NIC is not readable in AWS CI clusters. 

Here is what happens when reading the  `speed` file on a node in an AWS CI cluster.
```
drwxr-xr-x. 10 root root    0 Mar 14 11:22 queues
-r--r--r--.  1 root root 4096 Mar 14 11:20 speed
drwxr-xr-x.  2 root root    0 Mar 14 11:22 statistics
lrwxrwxrwx.  1 root root    0 Mar 14 11:20 subsystem -> ../../../../../class/net
-r--r--r--.  1 root root 4096 Mar 14 11:23 testing
-rw-r--r--.  1 root root 4096 Mar 14 11:23 threaded
-rw-r--r--.  1 root root 4096 Mar 14 11:23 tx_queue_len
-r--r--r--.  1 root root 4096 Mar 14 11:20 type
-rw-r--r--.  1 root root 4096 Mar 14 11:20 uevent
lrwxrwxrwx.  1 root root    0 Mar 14 11:22 upper_ovs-system -> ../../../../virtual/net/ovs-system
sh-4.4$ pwd
/host/root/sys/class/net/ens5
sh-4.4$ cat speed
cat: speed: Invalid argument
```

**Comment by simonpasquier** [comment]
@raptorsun this needs a backport on `release-4.13`

File: pkg/manifests/manifests.go (hunk 1 above)
Comment (simonpasquier) [reply pkg/manifests/manifests.go:2213]: IIUC an empty `argName` can't be triggered by user input (e.g. via CMO configmap) so I think that it's better to fail. If we hit this situation, it means that we have introduced a bug in the CMO logic.

```suggestion
			(argName[len(argName)-1] == '=' && strings.HasPrefix(arg, argName)) {
```

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Pull request merged**